apiserver/server.go

```diff
@@ -0,0 +1,161 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/astanishevskyi/http-server/apiserver/adapters"
+	"github.com/astanishevskyi/http-server/apiserver/configs"
+	"github.com/astanishevskyi/http-server/apiserver/models"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type Server struct {
+	config  *configs.Config
+	storage models.UserInterface
+}
+
+func New(config *configs.Config) *Server {
+	return &Server{
+		config: config,
+	}
+}
+
+func (s *Server) Run() error {
+	log.Println("Server is running on " + s.config.BindAddr)
+	return http.ListenAndServe(s.config.BindAddr, nil)
+}
+
+func (s *Server) ConfigStorage() error {
+	if s.config.Storage == "in-memory" {
+		log.Println("Storage is in-memory")
+		inMemoryStorage := adapters.NewInMemoryUserStorage()
+		s.storage = inMemoryStorage
+		return nil
+
+	}
+	return errors.New("no storage is set")
+}
+
+func (s *Server) ConfigRouter() {
+	http.HandleFunc("/user/", s.User)
+}
+
+func (s *Server) User(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+
+		id, _ := strconv.ParseUint(strings.TrimPrefix(r.URL.Path, "/user/"), 0, 32)
+
+		if id == 0 {
```

Comment:
If you don't want to return all users on requests like `/user/gibberish`, use some other logic. It is expected to return an error (400, 404, or other) instead.

Reply:
Fixed